lapce-ui/src/editor.rs

```diff
@@ -1890,62 +1891,159 @@ impl Widget<LapceTabData> for LapceEditor {
                 }
             }
             Event::Command(cmd) if cmd.is(LAPCE_UI_COMMAND) => {
-                let cmd = cmd.get_unchecked(LAPCE_UI_COMMAND);
-                if let LapceUICommand::ShowCodeActions(point) = cmd {
-                    let editor_data = data.editor_view_content(self.view_id);
-                    if let Some(actions) = editor_data.current_code_actions() {
-                        if !actions.is_empty() {
-                            let mut menu = druid::Menu::new("");
-
-                            for action in actions.iter() {
-                                let title = match action {
-                                    CodeActionOrCommand::Command(c) => {
-                                        c.title.clone()
-                                    }
-                                    CodeActionOrCommand::CodeAction(a) => {
-                                        a.title.clone()
-                                    }
-                                };
-                                let mut item = druid::MenuItem::new(title);
-                                item = item.command(Command::new(
-                                    LAPCE_UI_COMMAND,
-                                    LapceUICommand::RunCodeAction(action.clone()),
-                                    Target::Widget(editor_data.view_id),
-                                ));
-                                menu = menu.entry(item);
+                match cmd.get_unchecked(LAPCE_UI_COMMAND) {
+                    LapceUICommand::ShowCodeActions(point) => {
+                        let editor_data = data.editor_view_content(self.view_id);
+                        if let Some(actions) = editor_data.current_code_actions() {
+                            if !actions.is_empty() {
```

Comment:
I think returning here if empty might end up looking better.

Reply:
This is not part of the PR, I think it got here because of cargo fmt. 
I could still add an early return, you tell me.